app/src/main/java/com/gruppo4/audioUtility/AudioUtilityManager.java

```diff
@@ -0,0 +1,173 @@
+        );
+
+
+    }
+
+    /**
+     * Sets up the stream Volume to its maximum value (in percentage).
+     *
+     * @param context The current Context.
+     * @param stream  The chosen stream (should be ALARM, RING or MUSIC).
+     */
+    public static void setMaxVolume(Context context, AUMStream stream) {
+        setVolume(context, stream, MAX_PERCENTAGE);
+    }
+
+    /**
+     * Sets up the Stream Volume to its minimum value (in percentage).
+     *
+     * @param context The current Context.
+     * @param stream  The chosen stream (should be ALARM, RING or MUSIC).
+     */
+    public static void setMinVolume(Context context, AUMStream stream) {
+        setVolume(context, stream, MIN_PERCENTAGE);
+    }
+
+    /**
+     * @param stream The chosen Stream (should be ALARM, RING or MUSIC, otherwise an exception is thrown).
+     * @return The constant representing the Stream.
+     * @throws IllegalArgumentException if stream is not valid (It shouldn't happen, thanks to the enum type).
+     */
+    private static int getStream(AUMStream stream) throws IllegalArgumentException {
+
+        switch (stream) {
+            case ALARM:
+                return AudioManager.STREAM_ALARM;
+            case RING:
+                return AudioManager.STREAM_RING;
+            case MUSIC:
+                return AudioManager.STREAM_MUSIC;
+            default:
+                throw new IllegalArgumentException("\nIllegal Stream. It should be ALARM, RING or MUSIC .");
+        }
+    }
+
+    /**
+     * @param context The current Context.
+     * @return The current Vibrator instance.
+     */
+    protected static Vibrator getVibrator(Context context) {
+        return (Vibrator) context.getSystemService(Context.VIBRATOR_SERVICE);
+    }
+
+    /**
+     * Makes the target device vibrate.
+     *
+     * @param context The current Context.
+     */
+    public static void singleVibrate(Context context) {
+        int time = VIBRATION_TIME + VIBRATION_DELAY;
+        // Check if API version is >26 or not
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            getVibrator(context).vibrate(VibrationEffect.createOneShot(time, VibrationEffect.DEFAULT_AMPLITUDE));
+        } else {
+            //deprecated in API 26
+            getVibrator(context).vibrate(time);
+        }
+    }
+
+    /**
+     * Makes the target device vibrate.
+     *
+     * @param context The current Context.
+     */
+    public static void multipleVibrate(Context context) {
+        // Vibration pattern
+        long[] x = {VIBRATION_DELAY, VIBRATION_TIME, VIBRATION_DELAY, VIBRATION_TIME};
+        // Check if API version is >26 or not
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            getVibrator(context).vibrate(VibrationEffect.createWaveform(x, VIBRATION_REPEAT_CODE));
+        } else {
+            //deprecated in API 26
+            getVibrator(context).vibrate(x, VIBRATION_REPEAT_CODE);
+        }
+    }
+
```

Comment:
Since the class handles starting the vibration, it would make sense to have a method to stop it.

Reply:
There is indeed the method to stop it... But it's in MainActivity.
You are definitely right, though.. I should put this in the utility class.

Reply:
I agree with the @RicDeZen remark.